tracer/src/Datadog.Trace/TraceContext.cs

```diff
@@ -170,6 +170,10 @@ private static void AddSamplingPriorityTags(Span span, int samplingPriority)
 
         private void PropagateMetadata(ArraySegment<Span> spans)
         {
+            // When receiving chunks, the backend checks whether the aas.resource.id tag is present on any span to decide
+            // which metric to emit (datadog.apm.host.instance or datadog.apm.azure_resource_instance one).
+            DecorateRootSpan(spans.Array[0]);
```

Comment:
I don't think we can be certain this is the root span, I think you need something like this (similar to what we do in `AgentWriter`:
```suggestion
            DecorateRootSpan(spans.Array[0].Context.TraceContext?.RootSpan;);
```

Reply:
🤔 We don't need the root span to be decorated with this, any span sent to the intake will do. And in the case of PartialFllush, it's not sure we will be sending a root span.
Still your comment apply, but I believe I shall rename the `DecorateRootSpan` method to `DecorateSpan` or  `DecorateSpanWIthAASMetadata` until someone needs to add something else and thus have to find a better name. WDYT?

Reply:
Ah, yeah, I assumed it had to be the root Span because of the name 😄 No need for the change then. `DecorateWithAASMetadata` or similar works for me - like you say, we can rename if needs be 🙂

Reply:
ok fixed in [bcf191c](https://github.com/DataDog/dd-trace-dotnet/pull/2960/commits/bcf191c1dba3fc3beb9d0d83baf73f0647bf88bf)